lib/io/can.cpp

```diff
@@ -1,8 +1,79 @@
 #include "can.hpp"
 
+#include <stdio.h>
+#include <unistd.h>
+
+#include <linux/can.h>
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+
 namespace hyped::io {
-Can::Can()
+Can::Can(hyped::core::ILogger &logger) : logger_(logger)
+{
+}
+
+CanResult Can::initialiseCanSocket(std::string can_network_interface)
+{
+  socket_ = socket(PF_CAN, SOCK_RAW, CAN_RAW);
+  if (socket_ < 0) {
+    logger_.log(hyped::core::LogLevel::kFatal, "Unable to open CAN socket");
+    return hyped::io::CanResult::kFailure;
+  }
+  sockaddr_can socket_address;
+  socket_address.can_family  = AF_CAN;
+  socket_address.can_ifindex = if_nametoindex(can_network_interface.c_str());
+  if (socket_address.can_ifindex == 0) {
+    logger_.log(hyped::core::LogLevel::kFatal, "Unable to find CAN1 network interface");
+    close(socket_);
+    socket_ = -1;
+    return hyped::io::CanResult::kFailure;
+  }
+  const int bind_status
+    = bind(socket_, reinterpret_cast<sockaddr *>(&socket_address), sizeof(socket_address));
+  if (bind_status < 0) {
+    logger_.log(hyped::core::LogLevel::kFatal, "Unable to bind CAN socket");
+    close(socket_);
+    socket_ = -1;
+    return hyped::io::CanResult::kFailure;
+  }
+  logger_.log(hyped::core::LogLevel::kInfo, "CAN socket successfully created");
+  return hyped::io::CanResult::kSuccess;
+}
+
+CanResult Can::sendCanFrame(const core::CanFrame message)
```

Comment:
Why does `CanFrame` live in `core` rather than `io`? I'm not saying that its necessarily bad but we want to have some good heuristics for what goes where? @kshxtij

Reply:
But `io/` will be shared between the two?

Reply:
That is fair, @TomLonergan03 could you move the definition to can.hpp?